docs/csharp/programming-guide/concepts/expression-trees/how-to-use-expression-trees-to-build-dynamic-queries.md

```diff
@@ -1,107 +1,266 @@
-    typeof(Queryable),  
-    "OrderBy",  
-    new Type[] { queryableData.ElementType, queryableData.ElementType },  
-    whereCallExpression,  
-    Expression.Lambda<Func<string, string>>(pe, new ParameterExpression[] { pe }));  
-// ***** End OrderBy *****  
-  
-// Create an executable query from the expression tree.  
-IQueryable<string> results = queryableData.Provider.CreateQuery<string>(orderByCallExpression);  
-  
-// Enumerate the results.  
-foreach (string company in results)  
-    Console.WriteLine(company);  
-  
-/*  This code produces the following output:  
-  
-    Blue Yonder Airlines  
-    City Power & Light  
-    Coho Winery  
-    Consolidated Messenger  
-    Graphic Design Institute  
-    Humongous Insurance  
-    Lucerne Publishing  
-    Northwind Traders  
-    The Phone Company  
-    Wide World Importers  
-*/  
-```  
-  
- This code uses a fixed number of expressions in the predicate that is passed to the `Queryable.Where` method. However, you can write an application that combines a variable number of predicate expressions that depends on the user input. You can also vary the standard query operators that are called in the query, depending on the input from the user.  
-  
-## Compiling the Code  
-  
-- Include the System.Linq.Expressions namespace.  
-  
+# Querying based on runtime state (C#)
+
+Consider code that defines an <xref:System.Linq.IQueryable%601> or an <xref:System.Linq.IQueryable> against a data source:
+
+```csharp
+var companyNames = new[] {
+    "Consolidated Messenger", "Alpine Ski House", "Southridge Video",
+    "City Power & Light", "Coho Winery", "Wide World Importers",
+    "Graphic Design Institute", "Adventure Works", "Humongous Insurance",
+    "Woodgrove Bank", "Margie's Travel", "Northwind Traders",
+    "Blue Yonder Airlines", "Trey Research", "The Phone Company",
+    "Wingtip Toys", "Lucerne Publishing", "Fourth Coffee"
+};
+
+// We're using an in-memory array as the data source, but the IQueryable could have come
+// from anywhere -- an ORM backed by a database, a web request, or any other LINQ provider.
+IQueryable<string> companyNamesSource = companyNames.AsQueryable();
+
+var qry = companyNames.OrderBy(x => x);
+```
+
+Every time you run this code, the same exact query will be executed. This is frequently not very useful, as you may want your code to execute different queries under varying conditions. This topic describes how you can execute a different query based on runtime state.
+
+## IQueryable, IQueryable\<T> and expression trees
+
+Fundamentally, an <xref:System.Linq.IQueryable> has two components:
+
+* <xref:System.Linq.IQueryable.Expression>&mdash;a language- and datasource-agnostic representation of the current query's elements, in the form of an expression tree.
+* <xref:System.Linq.IQueryable.Provider>&mdash;an object which knows how to translate the current query into actual .NET objects, a.k.a. an instance of a LINQ provider.
```

Comment:
I'd leave off "into actual .NET objects", and avoid "a.k.a":

```suggestion
* <xref:System.Linq.IQueryable.Provider>&mdash;an object which knows how to translate the current query. The provider is an instance of a LINQ provider.
```

Reply:
Fixed in cc6550ea6ce39020e8838d4ff08fb2fb59232a73